src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1447,27 +1461,95 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
                                 {
-                                    this.StateWhenTrue[slot] = true;
+                                    state[slot] = true;
                                 }
                             }
                         }
                     }
                 }
             }
+
+            void getOperandSlots(BoundExpression operand, ArrayBuilder<int> slotListBuilder)
+            {
+                Debug.Assert(operand != null);
+                Debug.Assert(_lastConditionalAccessSlot == -1);
+                int slot;
+
+                do
+                {
+                    // Due to the nature of binding, if there are conditional access they will be at the top of the bound tree,
+                    // potentially with a conversion on top of it. We go through any conditional accesses, adding slots for the
+                    // conditional receivers if they have them. If we ever get to a receiver that MakeSlot doesn't return a slot
+                    // for, nothing underneath is trackable and we bail at that point. Example:
+                    //
+                    //     a?.GetB()?.C // a is a field, GetB is a method, and C is a property
+                    //
+                    // The top of the tree is the a?.GetB() conditional call. We'll ask for a slot for a, and we'll get one because
+                    // locals have slots. The AccessExpression of the BoundConditionalAccess is another BoundConditionalAccess, this time
+                    // with a receiver of the GetB() BoundCall. Attempting to get a slot for this receiver will fail, and we'll
+                    // return an array with just the slot for a.
+                    switch (operand.Kind)
+                    {
+                        case BoundKind.Conversion:
+                            // PROTOTYPE(NullableReferenceTypes): Detect when conversion has a nullable operand
+                            operand = ((BoundConversion)operand).Operand;
+                            continue;
+                        case BoundKind.ConditionalAccess:
+                            var conditional = (BoundConditionalAccess)operand;
+
+                            slot = MakeSlot(conditional.Receiver);
+                            if (slot > 0)
+                            {
+                                // If we got a slot we must have processed the previous conditional receiver.
+                                Debug.Assert(_lastConditionalAccessSlot == -1);
+
+                                slotListBuilder.Add(slot);
+
+                                // When MakeSlot is called on the nested AccessExpression, it will recurse through receivers
+                                // until it gets to the BoundConditionalReceiver associated with this node. In our override,
+                                // we substitute this slot when we encounter a BoundConditionalAccessReceiver, and reset the
```

Comment:
Should that be `BoundConditionalReceiver`?

Reply:
I keep getting if confused with `IConditionalAccessInstanceOperation`.

---
In reply to: [204186810](https://github.com/dotnet/roslyn/pull/28722#discussion_r204186810) [](ancestors = 204186810)